scripts/ci/assay_runner_lane_check.py

```
@@ -122,6 +148,20 @@ def request(self, method: str, path: str, payload: object | None = None) -> obje
 
+    def download(self, path_or_url: str) -> bytes:
+        url = path_or_url if path_or_url.startswith("https://") else f"{self.base_url}{path_or_url}"
+        request = urllib.request.Request(
+            url,
+            headers={
+                "Accept": "application/vnd.github+json",
+                "X-GitHub-Api-Version": "2022-11-28",
+                "Authorization": f"Bearer {self.token}",
+            },
+            method="GET",
+        )
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)